src/runtime/hexagon_remote/known_symbols.cpp

```diff
@@ -121,47 +121,47 @@ void *get_known_symbol(const char *name) {
         {"__hexagon_modsi3", (char *)(&__hexagon_modsi3)},
         {"__hexagon_sqrtf", (char *)(&sqrtf)},
         {"sqrtf", (char *)(&sqrtf)},
-        {"sqrt", (char *)(&sqrt)},
+        {"sqrt", (char *)(static_cast<double (*)(double)>(&sqrt))},
```

Comment:
Why are all these casts necessary?

Reply:
I'm guessing the newer Hexagon SDK headers are apparently including math.h transitively and were not before, and so now sqrt is an overloaded symbol and needs to be resolved to the right thing. sqrtf isn't overloaded so it doesn't need the cast.

Reply:
Changing the above:
```
-        {"sqrt", (char *)(static_cast<double (*)(double)>(&sqrt))},
+        {"sqrt", (char *)(&sqrt)},
```
and rebuilding, results in:

```
known_symbols.cpp:124:28: error: address of overloaded function 'sqrt' does not match required type 'char'
        {"sqrt", (char *)(&sqrt)},
                           ^~~~
known_symbols.cpp:36:15: note: candidate function
extern double sqrt(double);
              ^
/local/scratch/dpalermo/TSDK/Hexagon_SDK/4.3.0.0/tools/HEXAGON_Tools/8.4.11/Tools/bin/../target/hexagon/include/c++/v1/math.h:1075:46: note: candidate function
inline _LIBCPP_INLINE_VISIBILITY long double sqrt(long double __lcpp_x) _NOEXCEPT {return ::sqrtl(__lcpp_x);}
                                             ^
/local/scratch/dpalermo/TSDK/Hexagon_SDK/4.3.0.0/tools/HEXAGON_Tools/8.4.11/Tools/bin/../target/hexagon/include/c++/v1/math.h:1074:46: note: candidate function
inline _LIBCPP_INLINE_VISIBILITY float       sqrt(float __lcpp_x) _NOEXCEPT       {return ::sqrtf(__lcpp_x);}
                                             ^
/local/scratch/dpalermo/TSDK/Hexagon_SDK/4.3.0.0/tools/HEXAGON_Tools/8.4.11/Tools/bin/../target/hexagon/include/c++/v1/math.h:1081:1: note: candidate template ignored: could not match 'typename std::enable_if<std::is_integral<_A1>::value, double>::type (_A1) noexcept' against 'char'
sqrt(_A1 __lcpp_x) _NOEXCEPT {return ::sqrt((double)__lcpp_x);}
^
1 error generated.
```

Reply:
FWIW transitively including math.h from standard headers might be worth looking into if that is happening. If that's not standard behavior, it could cause problems, and unnecessarily parsing/dead code stripping math.h on *everything* compiled for hexagon is unfortunate.

I don't think any of these headers should be including math.h:
- https://github.com/halide/Halide/blob/master/src/runtime/HalideRuntime.h#L5-L14
- https://github.com/halide/Halide/blob/master/src/runtime/hexagon_remote/known_symbols.cpp#L3-L7

I don't see where else it could be coming from.

Reply:
Adding -E to the compile line shows that it was included from stdlib.h.

```
/local/scratch/dpalermo/TSDK/Hexagon_SDK/4.3.0.0/tools/HEXAGON_Tools/8.4.11/Tools/bin/../target/hexagon/include/c++/v1/stdlib.h:
...
#ifdef __cplusplus
#include <math.h>
#endif  // __cplusplus

#endif  // _LIBCPP_STDLIB_H
```

Reply:
This is coming from LLVM libcxx/include/stdlib.h.   LLVM 10 has the include math.h:

     https://github.com/llvm/llvm-project/blob/release/10.x/libcxx/include/stdlib.h

Reply:
The include math.h goes away in LLVM 11:

    https://github.com/llvm/llvm-project/blob/release/11.x/libcxx/include/stdlib.h

and it looks like they swtiched to builtins for what is needed by stdlib.h:

```
inline _LIBCPP_INLINE_VISIBILITY float abs(float __lcpp_x) _NOEXCEPT {
  return __builtin_fabsf(__lcpp_x); // Use builtins to prevent needing math.h
}
```